x-pack/plugins/canvas/canvas_plugin_src/functions/common/joinRows.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { ExpressionFunction } from '../../../../../../src/legacy/core_plugins/interpreter/types';
+import { Datatable } from '../types';
+import { getFunctionErrors } from '../../strings/functions';
+
+interface Arguments {
+  column: string;
+  separator: string;
+  quote: string;
+}
+
+const escapeString = (data: string, quotechar: string): string => {
+  if (quotechar === undefined || quotechar === '') {
+    return data;
+  } else {
+    const regex = new RegExp(quotechar, 'g');
+    return data.replace(/\\/g, `\\\\`).replace(regex, `\\${quotechar}`);
+  }
+};
+
+export function joinRows(): ExpressionFunction<'joinRows', Datatable, Arguments, string> {
+  const errors = getFunctionErrors().getCell;
+  return {
+    name: 'joinRows',
+    type: 'string',
+    help: 'Join values from rows in a datatable into a string',
+    context: {
+      types: ['datatable'],
+    },
+    args: {
+      column: {
+        aliases: ['_'],
+        types: ['string'],
+        required: true,
+        help: 'Column to join values from',
+      },
+      separator: {
+        aliases: ['sep'],
+        types: ['string'],
+        default: ',',
+        help: 'Separator to use between row values',
+      },
+      quote: {
+        types: ['string'],
+        default: 'DEFAULT',
+        help: 'Quote character around values',
+      },
```

Comment:
Can we add an argument called `distinct` or `unique`? From the name of the function, I don't think it's expected that the duplicates are automatically removed before joining the values in the function. I'm not sure if it's useful to join all the values including the duplicates, but I think it'd be good to be explicit about it here.

```suggestion
      },
      distinct: {
        types: ['boolean'],
        default: true,
        help: 'Extract unique values from the column?'.
      }
```

Reply:
@w33ble suggested adding distinct earlier. In my example i get the distinct hosts in my query (not sure if that's always possible). Added argument.